app/models/host/discovered.rb

```diff
@@ -3,7 +3,7 @@
 class Host::Discovered < ::Host::Base
   include ScopedSearchExtensions
 
-  belongs_to :subnet
+  has_one :subnet, :through => :primary_interface
```

Comment:
I am actually deleting this line

    has_one :subnet, :through => :primary_interface

because it is defined in `host/base` and I see no reason to redefine it here.